src/rdkafka.c

```diff
@@ -1427,6 +1442,12 @@ rd_kafka_t *rd_kafka_new (rd_kafka_type_t type, rd_kafka_conf_t *app_conf,
 	if (rk->rk_conf.offset_commit_cb)
 		rk->rk_conf.enabled_events |= RD_KAFKA_EVENT_OFFSET_COMMIT;
 
+        rk->rk_controllerid = -1;
+
+        /* Admin client defaults */
+        rk->rk_conf.admin.request_timeout_ms =
+                RD_MIN(5*60*1000, rk->rk_conf.socket_timeout_ms);
```

Comment:
why not just socket_timeout_ms? (and why the magic constant?)

Reply:
I don't remember! Reverting :)